src/map.react.js

```diff
@@ -303,6 +305,20 @@ export default class MapGL extends Component {
     map.addSource(update.id, newSource);
   }
 
+  // Hover and click only query layers whose interactive property is true
+  // If no interactivity is specified, query all layers
+  _updateQueryParams(mapStyle) {
+    if (mapStyle instanceof Immutable.Map) {
+      mapStyle = mapStyle.toJS();
```

Comment:
Getting a little worried that as we add style manipulating features, we risk converting the entire mapStyle blob `toJS()` more than once. That would be the big performance hit here. Can we just extract the layers and convert just those?